### ISSUE TITLE
Kysley D1 Adaptor

### ISSUE THREAD

**Issue by tom-huntington** [james-elicx/cf-bindings-proxy#9]
The adaptor code is a simple wrapper https://github.com/aidenwallis/kysely-d1/blob/ba48112890fd91035fb51b93dc1301e9f8e4329d/src/index.ts#L98C25-L108

```
const d1 = binding<D1Database>('DB');
const kysley = new Kysely<DB>({ dialect: new D1Dialect({ database: d1 }) });
const boards_table = await kysley.selectFrom('Board').selectAll().execute();
```

The error only occurs when going through the proxy
> Error: async (...l) => (i.__calls.push({ prop: r, args: l }), c(i))
    at D1Connection.executeQuery (/home/a/repos/astroboard/node_modules/kysely-d1/dist/index.js:86:19)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async file:///home/a/repos/astroboard/node_modules/kysely/dist/esm/query-executor/query-executor-base.js:35:28
    at async DefaultConnectionProvider.provideConnection (file:///home/a/repos/astroboard/node_modules/kysely/dist/esm/driver/default-connection-provider.js:10:20)
    at async DefaultQueryExecutor.executeQuery (file:///home/a/repos/astroboard/node_modules/kysely/dist/esm/query-executor/query-executor-base.js:34:16)
    at async SelectQueryBuilder.execute (file:///home/a/repos/astroboard/node_modules/kysely/dist/esm/query-builder/select-query-builder.js:1134:24)
    at async eval (/home/a/repos/astroboard/src/pages/[board]/index.astro:23:22)
    at async renderPage (file:///home/a/repos/astroboard/node_modules/astro/dist/runtime/server/render/page.js:96:30)
    at async renderPage (file:///home/a/repos/astroboard/node_modules/astro/dist/core/render/core.js:41:18)
    at async renderPage (file:///home/a/repos/astroboard/node_modules/astro/dist/core/render/dev/index.js:127:10)
>
>async executeQuery(compiledQuery) {
        // Transactions are not supported yet.
        // if (this.#transactionClient) return this.#transactionClient.executeQuery(compiledQuery)
        const results = await __classPrivateFieldGet(this, _D1Connection_config, "f").database
            .prepare(compiledQuery.sql)
            .bind(...compiledQuery.parameters)
            .all();
        if (results.error) {
            throw new Error(results.error);
            ^
        }

works fine if I implement the function myself
```
const compiledQuery = kysley.selectFrom('Board').selectAll().compile();
console.log(compiledQuery, "<------------ compiledQuery");
const results = await d1
            .prepare(compiledQuery.sql)
            .bind(...compiledQuery.parameters)
            .all();
console.log(results, "<-------------- results");
```

Have any idea where this is coming from? `Error: async (...l) => (i.__calls.push({ prop: r, args: l }), c(i))`

**Comment by tom-huntington** [comment]
So the problem is that the `if(results.error)` is path always takes as it evaluates to the proxy function
```
    const results = await this.#config.database
      .prepare(compiledQuery.sql)
      .bind(...compiledQuery.parameters)
      .all();
     console.log(results.error, "<----------results.error")
      if (results.error) {
        console.log(results, "<----------results.error")
```

```
[proxy]: [AsyncFunction (anonymous)] <----------results.error
[proxy]: {
  results: [ { id: 1, name: 'b' } ],
  success: true,
  meta: {
    duration: 0.26789999986067414,
    last_row_id: 0,
    changes: 0,
    served_by: 'miniflare.db',
    internal_stats: null
  }
} <----------results.error
```

Here's the type of results
```
declare interface D1Result<T = unknown> {
  results: T[];
  success: true;
  meta: any;
  error?: never;
}
```

**Comment by james-elicx** [comment]
Thank you for the additional information Tom. I can see what's causing this. When the D1Result properties aren't defined, it wouldn't find the property in the data, so it fell back to returning a new proxy function which is primarily useful for r2.

I have opened a PR to fix this so that it can return undefined for those specific properties when they don't exist. I have also added a test using Kysely too.

Sorry about the problem.
